generator/src/main/java/org/corfudb/generator/State.java

```diff
@@ -0,0 +1,110 @@
+package org.corfudb.generator;
+
+import com.google.common.reflect.TypeToken;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import org.corfudb.generator.distributions.Keys;
+import org.corfudb.generator.distributions.OperationCount;
+import org.corfudb.generator.distributions.Operations;
+import org.corfudb.generator.distributions.Streams;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.collections.SMRMap;
+import org.corfudb.runtime.object.transactions.TransactionType;
+
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * This object keeps state information of the different data distributions and runtime client.
+ *
+ * Created by maithem on 7/14/17.
+ */
+public class State {
+
+    @Getter
+    final Streams streams;
+
+    @Getter
+    final Keys keys;
+
+    @Getter
+    final OperationCount operationCount;
+
+    @Getter
+    final Operations operations;
+
+    @Getter
+    final CorfuRuntime runtime;
+
+    @Getter
+    final Map<UUID, SMRMap> maps;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "SMRMap"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)